modules/generator/registry/counter.go

```diff
@@ -27,11 +27,35 @@ type counterSeries struct {
 	labelValues []string
 	value       *atomic.Float64
 	lastUpdated *atomic.Int64
+	// firstSeries is used to track if this series is new to the counter.  This
+	// is used to ensure that new counters being with 0, and then are incremented
+	// to the desired value.  This avoids Prometheus throwing away the first
+	// value in the series, due to the transition from null -> x.
+	firstSeriesMtx sync.RWMutex
+	firstSeries    bool
```

Comment:
Is there a specific reason to use `sync.RWMutex` over `atomic.Bool`? I'm not entirely sure which solution would be more performant, but if it's equal I'd say `atomic.Bool` is easier to work with.

An alternative approach would be to store a `created` timestamp. `isNew()` could then be implemented as `created == lastUpdated.load()`.
Since `created` would never be modified we don't need any concurrency handling for it.

Reply:
I'd considered the alternative you mentioned here, which seems a little cleaner than any locking.  I'll revisit this.

Reply:
Seems a little cleaner after the update.  I think we still want to be able to turn off the isNew, and so kept the bool in place of the timestamp approach.